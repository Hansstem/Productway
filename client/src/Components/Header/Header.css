@import url('https://fonts.googleapis.com/css2?family=Montserrat:wght@400;700&display=swap');

.Header {
	background-color: #354f52;
	display: flex;
	flex-direction: row;
	align-items: center; /* Center vertically */
	font-size: calc(4px + 0.8vmin);
	color: white;
	font-family: 'Montserrat', sans-serif;
	user-select: none;
}

.header-icon {
	cursor: pointer;
	margin-left: 20px; /* Adjust margin as needed */
}

.header-title {
	cursor: pointer;
	margin-left: 10px; /* Adjust margin as needed */
}
.header-spacer {
	margin-left: auto;
	margin-right: 4em;
	display: flex;
	flex-direction: row;
	align-items: center; /* Center vertically */
}
.search-and-button {
	display: flex;
	align-items: center;
	height: 100%; /* Make it the same height as the Header */
}

@media screen and (max-width: 650px) {
	.header-spacer {
		display: none;
	}

	.search-and-button {
		margin-left: auto;
	}
}
