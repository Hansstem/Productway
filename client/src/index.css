:root {
	font-family: Inter, system-ui, Avenir, Helvetica, Arial, sans-serif;
	line-height: 1.5;
	font-weight: 400;

	color-scheme: light dark;
	color: #333333;
	background-color: #242424;

	font-synthesis: none;
	text-rendering: optimizeLegibility;
	-webkit-font-smoothing: antialiased;
	-moz-osx-font-smoothing: grayscale;
	-webkit-text-size-adjust: 100%;
}

#root {
	width: 100%;
	max-width: 100%;
	box-sizing: border-box; /* This ensures padding and border are included in the width */
}

a {
	font-weight: 500;
	color: #646cff;
	text-decoration: inherit;
}
a:hover {
	color: #535bf2;
}

body {
	margin: 0;
	display: flex;
	min-height: 100vh;
	background-color: #f5f5f5;
}

h1 {
	font-size: 3.2em;
	line-height: 1.1;
}

button {
	margin: 20px;
	background-color: #f8f8f8;
	color: black;
	border: 1px solid #ddd;
	padding: 10px 20px;
	border-radius: 15px;
	box-shadow: 0px 2px 3px rgba(0, 0, 0, 0.1);
	font-size: 16px;
	transition: all 0.3s ease;
	cursor: pointer;
}

button:hover {
	background-color: #f8f8f8; /* Slightly darker background on hover */
	box-shadow: 0px 2px 4px rgba(0, 0, 0, 0.15); /* More pronounced shadow on hover */
	transform: translateY(-0.5px); /* Slight raise effect on hover */
}

button:active {
	transform: translateY(1px); /* Button appears pressed when active */
}

@media (prefers-color-scheme: light) {
	:root {
		color: #213547;
		background-color: #ffffff;
	}
}
