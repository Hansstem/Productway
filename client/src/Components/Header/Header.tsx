import './Header.css';
import { useNavigate } from 'react-router-dom';
import headerIcon from '../../assets/header-icon.svg';
import Searchbar from '../Searchbar/Searchbar.tsx';
import { IProduct } from '../../data/types';

export default function Header({ products }: { products: IProduct[] }) {
	const navigate = useNavigate();

	function handleNavHome() {
		navigate('/');
	}

	return (
		<div className="Header">
			<img src={headerIcon} alt="Header Icon" className="header-icon" />{' '}
			{/* Use the imported SVG */}
			<h1 onClick={() => handleNavHome()} className="header-title">
				{' '}
				PRODUCTWAY
			</h1>
			<div className="header-spacer">
				<Searchbar products={products} />
			</div>
		</div>
	);
}
