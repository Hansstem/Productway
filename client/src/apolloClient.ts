import { ApolloClient, InMemoryCache } from '@apollo/client';

export const client = new ApolloClient({
	uri: 'http://it2810-10.idi.ntnu.no:4000/graphql',
	credentials: 'include',
	cache: new InMemoryCache(),
});
