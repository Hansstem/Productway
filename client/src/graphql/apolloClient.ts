import { ApolloClient, InMemoryCache } from '@apollo/client';

export const client = new ApolloClient({
	uri: 'http://it2810-10.idi.ntnu.no:4000/graphql', // for production
	//uri: 'http://localhost:4000/graphql', // for local development
	credentials: 'include',
	cache: new InMemoryCache(),
});
