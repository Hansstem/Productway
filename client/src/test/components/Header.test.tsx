import { StrictMode } from 'react';

import { ApolloProvider } from '@apollo/client';
import { render, RenderResult, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { RecoilRoot } from 'recoil';

import { client } from '../../apolloClient';
import Header from '../../Components/Header/Header';

describe('Header test', () => {
	let header: RenderResult;
	beforeEach(() => {
		header = render(
			<StrictMode>
				<RecoilRoot>
					<ApolloProvider client={client}>
						<MemoryRouter>
							<Header />
						</MemoryRouter>
					</ApolloProvider>
				</RecoilRoot>
			</StrictMode>,
		);
	});

	it('snapshottest seader', () => {
		expect(header).toMatchSnapshot();
	});

	it('renders correctly', () => {
		expect(screen.getAllByText(/Productway/i)).toBeTruthy();
		expect(screen.getAllByText(/Min side/i)).toBeTruthy();
		expect(
			screen.getAllByPlaceholderText(/SÃ¸k etter produkter.../i),
		).toBeTruthy();
	});
});
