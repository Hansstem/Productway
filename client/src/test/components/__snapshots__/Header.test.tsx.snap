// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header test > snapshottest header 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="Header"
      >
        <div
          class="desktopHeader"
        >
          <img
            alt="Header Icon"
            class="header-icon"
            src="/project2/src/assets/header-icon.svg"
          />
           
          <h1
            class="header-title"
          >
            PRODUCTWAY
          </h1>
          <div
            class="header-spacer"
          >
            <div
              class="searchBar"
            >
              <div
                class="MuiAutocomplete-root MuiAutocomplete-variantOutlined MuiAutocomplete-colorNeutral MuiAutocomplete-sizeMd css-10cfyqz-JoyAutocomplete-root"
                style="height: 3.5rem; background-color: white;"
              >
                <div
                  class="MuiAutocomplete-wrapper css-n43o-JoyAutocomplete-wrapper"
                >
                  <input
                    aria-autocomplete="list"
                    aria-expanded="false"
                    autocapitalize="none"
                    autocomplete="off"
                    class="MuiAutocomplete-input css-1epe1tr-JoyAutocomplete-input"
                    id=":r1:"
                    placeholder="Søk etter produkter..."
                    role="combobox"
                    spellcheck="false"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="search-and-button"
          >
            <div
              class="MyPageButton"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-q85h0t-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="button"
              >
                Min side
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
        <div
          class="mobileSearchBar"
        >
          <div
            class="mobileSpacer"
          >
            <div
              class="searchBar"
            >
              <div
                class="MuiAutocomplete-root MuiAutocomplete-variantOutlined MuiAutocomplete-colorNeutral MuiAutocomplete-sizeMd css-10cfyqz-JoyAutocomplete-root"
                style="height: 3.5rem; background-color: white;"
              >
                <div
                  class="MuiAutocomplete-wrapper css-n43o-JoyAutocomplete-wrapper"
                >
                  <input
                    aria-autocomplete="list"
                    aria-expanded="false"
                    autocapitalize="none"
                    autocomplete="off"
                    class="MuiAutocomplete-input css-1epe1tr-JoyAutocomplete-input"
                    id=":r3:"
                    placeholder="Søk etter produkter..."
                    role="combobox"
                    spellcheck="false"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="Header"
    >
      <div
        class="desktopHeader"
      >
        <img
          alt="Header Icon"
          class="header-icon"
          src="/project2/src/assets/header-icon.svg"
        />
         
        <h1
          class="header-title"
        >
          PRODUCTWAY
        </h1>
        <div
          class="header-spacer"
        >
          <div
            class="searchBar"
          >
            <div
              class="MuiAutocomplete-root MuiAutocomplete-variantOutlined MuiAutocomplete-colorNeutral MuiAutocomplete-sizeMd css-10cfyqz-JoyAutocomplete-root"
              style="height: 3.5rem; background-color: white;"
            >
              <div
                class="MuiAutocomplete-wrapper css-n43o-JoyAutocomplete-wrapper"
              >
                <input
                  aria-autocomplete="list"
                  aria-expanded="false"
                  autocapitalize="none"
                  autocomplete="off"
                  class="MuiAutocomplete-input css-1epe1tr-JoyAutocomplete-input"
                  id=":r1:"
                  placeholder="Søk etter produkter..."
                  role="combobox"
                  spellcheck="false"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="search-and-button"
        >
          <div
            class="MyPageButton"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-q85h0t-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              Min side
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        class="mobileSearchBar"
      >
        <div
          class="mobileSpacer"
        >
          <div
            class="searchBar"
          >
            <div
              class="MuiAutocomplete-root MuiAutocomplete-variantOutlined MuiAutocomplete-colorNeutral MuiAutocomplete-sizeMd css-10cfyqz-JoyAutocomplete-root"
              style="height: 3.5rem; background-color: white;"
            >
              <div
                class="MuiAutocomplete-wrapper css-n43o-JoyAutocomplete-wrapper"
              >
                <input
                  aria-autocomplete="list"
                  aria-expanded="false"
                  autocapitalize="none"
                  autocomplete="off"
                  class="MuiAutocomplete-input css-1epe1tr-JoyAutocomplete-input"
                  id=":r3:"
                  placeholder="Søk etter produkter..."
                  role="combobox"
                  spellcheck="false"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Header test > snapshottest seader 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="Header"
      >
        <div
          class="desktopHeader"
        >
          <img
            alt="Header Icon"
            class="header-icon"
            src="/project2/src/assets/header-icon.svg"
          />
           
          <h1
            class="header-title"
          >
            PRODUCTWAY
          </h1>
          <div
            class="header-spacer"
          >
            <div
              class="searchBar"
            >
              <div
                class="MuiAutocomplete-root MuiAutocomplete-variantOutlined MuiAutocomplete-colorNeutral MuiAutocomplete-sizeMd css-10cfyqz-JoyAutocomplete-root"
                style="height: 3.5rem; background-color: white;"
              >
                <div
                  class="MuiAutocomplete-wrapper css-n43o-JoyAutocomplete-wrapper"
                >
                  <input
                    aria-autocomplete="list"
                    aria-expanded="false"
                    autocapitalize="none"
                    autocomplete="off"
                    class="MuiAutocomplete-input css-1epe1tr-JoyAutocomplete-input"
                    id=":r1:"
                    placeholder="Søk etter produkter..."
                    role="combobox"
                    spellcheck="false"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="search-and-button"
          >
            <div
              class="MyPageButton"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-q85h0t-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="button"
              >
                Min side
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
        <div
          class="mobileSearchBar"
        >
          <div
            class="mobileSpacer"
          >
            <div
              class="searchBar"
            >
              <div
                class="MuiAutocomplete-root MuiAutocomplete-variantOutlined MuiAutocomplete-colorNeutral MuiAutocomplete-sizeMd css-10cfyqz-JoyAutocomplete-root"
                style="height: 3.5rem; background-color: white;"
              >
                <div
                  class="MuiAutocomplete-wrapper css-n43o-JoyAutocomplete-wrapper"
                >
                  <input
                    aria-autocomplete="list"
                    aria-expanded="false"
                    autocapitalize="none"
                    autocomplete="off"
                    class="MuiAutocomplete-input css-1epe1tr-JoyAutocomplete-input"
                    id=":r3:"
                    placeholder="Søk etter produkter..."
                    role="combobox"
                    spellcheck="false"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="Header"
    >
      <div
        class="desktopHeader"
      >
        <img
          alt="Header Icon"
          class="header-icon"
          src="/project2/src/assets/header-icon.svg"
        />
         
        <h1
          class="header-title"
        >
          PRODUCTWAY
        </h1>
        <div
          class="header-spacer"
        >
          <div
            class="searchBar"
          >
            <div
              class="MuiAutocomplete-root MuiAutocomplete-variantOutlined MuiAutocomplete-colorNeutral MuiAutocomplete-sizeMd css-10cfyqz-JoyAutocomplete-root"
              style="height: 3.5rem; background-color: white;"
            >
              <div
                class="MuiAutocomplete-wrapper css-n43o-JoyAutocomplete-wrapper"
              >
                <input
                  aria-autocomplete="list"
                  aria-expanded="false"
                  autocapitalize="none"
                  autocomplete="off"
                  class="MuiAutocomplete-input css-1epe1tr-JoyAutocomplete-input"
                  id=":r1:"
                  placeholder="Søk etter produkter..."
                  role="combobox"
                  spellcheck="false"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="search-and-button"
        >
          <div
            class="MyPageButton"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-q85h0t-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              Min side
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        class="mobileSearchBar"
      >
        <div
          class="mobileSpacer"
        >
          <div
            class="searchBar"
          >
            <div
              class="MuiAutocomplete-root MuiAutocomplete-variantOutlined MuiAutocomplete-colorNeutral MuiAutocomplete-sizeMd css-10cfyqz-JoyAutocomplete-root"
              style="height: 3.5rem; background-color: white;"
            >
              <div
                class="MuiAutocomplete-wrapper css-n43o-JoyAutocomplete-wrapper"
              >
                <input
                  aria-autocomplete="list"
                  aria-expanded="false"
                  autocapitalize="none"
                  autocomplete="off"
                  class="MuiAutocomplete-input css-1epe1tr-JoyAutocomplete-input"
                  id=":r3:"
                  placeholder="Søk etter produkter..."
                  role="combobox"
                  spellcheck="false"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
